src/components/ReactTrixRTEInput/ReactTrixRTEInput.jsx

```diff
@@ -35,27 +35,30 @@ function ReactTrixRTEInput(props) {
-
+    if (trixInputRef.current) {
+      trixRTEInputRef.current.addEventListener("trix-change", handleChange);
+      if(onFocus) trixRTEInputRef.current.addEventListener("trix-focus", onFocus);
+      if(onBlur) trixRTEInputRef.current.addEventListener("trix-blur", onBlur);
+      if(onInitialize) trixRTEInputRef.current.addEventListener("trix-initialize", onInitialize);
+      if(onFileAccepted) trixRTEInputRef.current.addEventListener("trix-file-accept", onFileAccepted);
+      if(onAttachmentAdd) trixRTEInputRef.current.addEventListener("trix-attachment-add", onAttachmentAdd);
+      if(onAttachmentRemove) trixRTEInputRef.current.addEventListener("trix-attachment-remove", onAttachmentAdd);
+      if(onSelectionChange) trixRTEInputRef.current.addEventListener("trix-selection-change", onSelectionChange);
+      if(onBeforeInitialize) trixRTEInputRef.current.addEventListener("trix-before-initialize", onBeforeInitialize);
+    }
 
     return () => {
-      trixRTEInputRef.current.removeEventListener("trix-change", handleChange);
-      if(onFocus) trixRTEInputRef.current.removeEventListener("trix-focus", onFocus);
-      if(onBlur) trixRTEInputRef.current.removeEventListener("trix-blur", onBlur);
-      if(onInitialize) trixRTEInputRef.current.removeEventListener("trix-initialize", onInitialize);
-      if(onFileAccepted) trixRTEInputRef.current.removeEventListener("trix-file-accept", onFileAccepted);
-      if(onAttachmentAdd) trixRTEInputRef.current.removeEventListener("trix-attachment-add", onAttachmentAdd);
-      if(onSelectionChange) trixRTEInputRef.current.removeEventListener("trix-selection-change", onSelectionChange);
-      if(onAttachmentRemove) trixRTEInputRef.current.removeEventListener("trix-attachment-remove", onAttachmentAdd);
-      if(onBeforeInitialize) trixRTEInputRef.current.removeEventListener("trix-before-initialize", onBeforeInitialize);
+      if (trixInputRef.current) {
+        trixRTEInputRef.current.removeEventListener("trix-change", handleChange);
+        if(onFocus) trixRTEInputRef.current.removeEventListener("trix-focus", onFocus);
+        if(onBlur) trixRTEInputRef.current.removeEventListener("trix-blur", onBlur);
+        if(onInitialize) trixRTEInputRef.current.removeEventListener("trix-initialize", onInitialize);
+        if(onFileAccepted) trixRTEInputRef.current.removeEventListener("trix-file-accept", onFileAccepted);
+        if(onAttachmentAdd) trixRTEInputRef.current.removeEventListener("trix-attachment-add", onAttachmentAdd);
+        if(onSelectionChange) trixRTEInputRef.current.removeEventListener("trix-selection-change", onSelectionChange);
+        if(onAttachmentRemove) trixRTEInputRef.current.removeEventListener("trix-attachment-remove", onAttachmentAdd);
+        if(onBeforeInitialize) trixRTEInputRef.current.removeEventListener("trix-before-initialize", onBeforeInitialize);
+      }
     };
   }, [])
```

Comment:
I was about to create a PR exactly for this issue. Using NextJS 10 / React 17 with functional refs, not using any custom refs, will get a `Cannot read property 'removeEventListener' of null` when the component unmounts.

This could be further improved by forcing the effect to conditionally update when `trixRTEInputRef.current` changes and not returning a callback when the ref is undefined. E.g.

```
useEffect(() => {
  if (trixRTEInputRef.current) {
      // addEventListeners
      
      return () => {
         // removeEventListeners
      };
    }
  },
  [trixRTEInputRef.current]
);

Reply:
@CUnknown From the above comment of @gleuch , could you also add `[trixRTEInputRef.current]` to `useEffect`?

```
useEffect(() => {
  ...
}, [trixRTEInputRef.current])
```